protodroid/src/main/java/id/lukasdylan/grpc/protodroid/internal/repository/InternalProtodroidRepository.kt

```diff
@@ -4,17 +4,71 @@ import id.lukasdylan.grpc.protodroid.internal.database.ProtodroidDao
 import id.lukasdylan.grpc.protodroid.internal.database.ProtodroidDataEntity
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.withContext
 
 internal interface InternalProtodroidRepository {
     suspend fun fetchAllData(): Flow<List<ProtodroidDataEntity>>
     suspend fun fetchSingleData(dataId: Long): Flow<ProtodroidDataEntity>
     suspend fun deleteAllData()
+    suspend fun getFilteredData(): Flow<List<ProtodroidDataEntity>>
+    fun updateFilter(filterType: FilterType)
+
+    val filterList: MutableList<FilterType>
```

Comment:
I think no need to put this variable on this interface. Since its only used on MainScreen, why dont you put on the ViewModel?? I checked on ViewModel also you declare again another variable which is a redundant for me

Reply:
yup make sense, will do that